src/simple_ai/models.py

```
@@ -114,7 +114,7 @@ def get_model(model_id: str, metadata: dict = MODELS_ZOO, task: str = "complete"
     if model_id in metadata.keys():
         model_interface = metadata.get(model_id).get("network", dict())
         model_url = model_interface.get("url", None)
-        model_interface = model_interface.get("type", None)
```

I'm not sure the changes there are for the better if that's driven by a config file issue. 

I think it might be bringing more clarity to expect from the user to explicit which kind of interface we're using (so far it's only gRPC but I would like to add at least REST later on). Maybe useful to have some default settings to allow leaner configuration though.

---

```

def select_model_type(model_interface: str, task: str = "complete"):
    if model_interface == "gRPC":
        if task == "embed":
            return RpcEmbeddingLanguageModel
        if task == "chat":
            return RpcChatLanguageModel
        return RpcCompletionLanguageModel
    return RpcCompletionLanguageModel


def get_model(model_id: str, metadata: dict = MODELS_ZOO, task: str = "complete"):
    if model_id in metadata.keys():
        model_interface = metadata.get(model_id).get("network", dict())
        model_url = model_interface.get("url", None)
        model_interface = model_interface.get("type", "gRPC")
        return select_model_type(model_interface, task)(name=model_id, url=model_url)
    else:
        return None
```
the default is set here,  `model_interface = model_interface.get("type", "gRPC")`. the explicit `None` makes it though into the function doesn't get replaced. should maybe be `model_type = ` too?

---

The `None` / `"gRPC"` is the default value if the `"type"` field isn't present in `models.toml` for the given model. But as long as it is there and using a valid value (`"gRPC"`), it works as intended in the original version. If everything is configured properly, neither the `None` or `"gRPC"` shall be accessed. 

I reckon that your change works too, I'm just not convinced it's necesary right now, but I might be missing your point. :)
Tbh that might be a change for the better to have default behaviours for missing fields, but that might be a discussion to have later outside the scope of this PR.

---

sorry that was not clear at all 

so there's two things I was trying to say
1) slightly confusing variable naming, `model_interface = metadata.get(model_id).get("network", dict())` gives the network configuration for the server and is declared as `model_interface` then later `model_interface = model_interface.get("type", "gRPC")` model interface is redeclared this type from the type of the model interface

2) since `model_interface.get("type", None)`  will either return a string or None, this gets passed into `select_model_type(model_interface, task)`, model_interface if undeclared defaults to `gRPC` however it is never undeclared so the default is never used and all calls to this function without explicit type set in configuration will fail the first conditional of `select_model_type` and automatically fall back to the completion model

eg we hit the `f(None)` situation when type is not declared
```
>>> def f(a=0):
...     return a
... 
>>> print(f(), f(None), f(1))
0 None 1
```
Will print `0 None 1`

> everything is configured properly, neither the None or "gRPC" shall be accessed.

is this suggesting to make type required in the future?

---

Sorry I've been unclear too, but:

1. Agree on **1**, it's a bit ugly and confusing and should be changed.
2. Yes I am indeed leaning towards making `type` mandatory, as I believe it would avoid confusion (since I would like to at least add support for REST soon) with implicit values. That's actually already part of `models.toml.template`.

> 2\. automatically fall back to the completion model

Yes, there is room for improvement here. Wouldn't it be better to then raise an error / return `None` to avoid this, rather than burying default values in the middle of `get_model` in `model_interface = model_interface.get("type", "gRPC")`?